weave/scorers/guardrails/restricted_terms_guardrail.py

```diff
@@ -0,0 +1,159 @@
+from typing import TYPE_CHECKING, Any, Optional, Union
+
+from pydantic import BaseModel, Field
+
+import weave
+from weave.scorers import Scorer
+from weave.scorers.guardrails.prompts import (
+    RESTRICTED_TERMS_GUARDRAIL_SYSTEM_PROMPT,
+    RESTRICTED_TERMS_GUARDRAIL_USER_PROMPT,
+)
+from weave.scorers.llm_utils import OPENAI_DEFAULT_MODEL, create
+
+if TYPE_CHECKING:
+    from instructor import Instructor
+
+
+class TermMatch(BaseModel):
+    """Represents a matched term and its variations"""
+
+    original_term: str
+    matched_text: str
+    match_type: str = Field(
+        description="Type of match: EXACT, MISSPELLING, ABBREVIATION, or VARIANT"
+    )
+    explanation: str = Field(
+        description="Explanation of why this is considered a match"
+    )
+
+
+class RestrictedTermsAnalysis(BaseModel):
+    """Analysis result for restricted terms detection"""
+
+    contains_restricted_terms: bool = Field(
+        description="Whether any restricted terms were detected"
+    )
+    detected_matches: list[TermMatch] = Field(
+        default_factory=list,
+        description="List of detected term matches with their variations",
+    )
+    explanation: str = Field(description="Detailed explanation of the analysis")
+    anonymized_text: Optional[str] = Field(
+        default=None,
+        description="Text with restricted terms replaced with category tags",
+    )
+
+    @property
+    def safe(self) -> bool:
```

Comment:
From Tim:
class RestrictedTermsAnalysis(BaseModel): 's safe property seems incorrect to me since safe should be a function of some input, not a property of the class